### PR TITLE
Fix GetMulitpleTorrents.user.js to work with v3.9

### PR DESCRIPTION
**Pull request by flatmapthatshit**
In order to make the script work with ruTorrent v3.9 I had to switch from GET to POST requests

### CODE REVIEW

**Comment by flatmapthatshit** [comment]
Tested with Chrome Version 75.0.3770.80 and Firefox 60.6.3esr.

By the way, while built-in functionality https://github.com/Novik/ruTorrent/pull/1796 is the way to go , however some providers have not updated the source plugin or have the necessary php modules to make it work.

**Comment by flatmapthatshit** [comment]
Hi @hannsen , see comment above. It is, but unfortunately not all providers have updated the plugin or PHP requirements (zip modules).

File: Rutorrent GetMulitpleTorrents.user.js [hunk 1]
```diff
@@ -27,17 +27,36 @@ if (nav_bar) {
...
+  let dl_link = 'plugins/source/action.php';
 
   if ('/' != window.location.toString().slice(-1)) {
     dl_link = '/' + dl_link;
   }
 
-  for (var i = 0; i < (selected.length - 1); i++) {
-    var hash = selected[i].id;
+  function downloadTorrent(hash) {
+    const form = document.createElement("form");
```
Comment (hannsen) [inline Rutorrent GetMulitpleTorrents.user.js:38]: Maybe this works too? Would be more direct

```
  let xhttp = new XMLHttpRequest();
  xhttp.open("POST", dl_link , true);
  // xhttp.setRequestHeader("Content-type", "application/x-www-form-urlencoded");   // maybe?
  xhttp.send("name=" + hash);
```

**Comment by hannsen** [comment]
Also did you test it with v3.8 too? Or only v3.9

File: Rutorrent GetMulitpleTorrents.user.js (hunk 1 above)
Comment (flatmapthatshit) [reply Rutorrent GetMulitpleTorrents.user.js:38]: Going with AJAX was also my first intuition. However, unfortunately, when using AJAX the response does not trigger the browser to download the torrent file by itself.
To make it work ultimately I would still need to create a link and trigger a fake click event or resort to other ugly workarounds with iframes or window.open.

Have a look at https://stackoverflow.com/questions/16086162/handle-file-download-from-ajax-post for further details.

Ultimately, while the form submission tech feels hacky, I couldn't come with a non-hacky solution.

**Comment by flatmapthatshit** [comment]
For now v3.9 only, although I expect it to work fine with v3.8 (on the PHP side of things, source was using `$_REQUEST` that should work fine with post parameters). Do you have a v3.8 installation easily available to test it? Otherwise I can probably spin up a docker container when I get home from work tomorrow.

**Comment by hannsen** [comment]
Thanks for your changes! (it worked on 3.8 too)

**Comment by flatmapthatshit** [comment]
Hi @hannsen , thank you very much for merging it.  If you could kindly also update https://greasyfork.org/en/scripts/29658-rutorrent-getmulitpletorrents that would be brilliant :).

**Comment by hannsen** [comment]
greasyfork auto updates scripts from git

**Comment by kdantas** [comment]
Hi @hannsen, is it possible download the multiples selected .torrents in one zip file?

**Comment by hannsen** [comment]
No that I know of, but rutorrent support downloading multiple torrents by itself now

**Pull request merged**